crates/core/src/topology.rs

```diff
@@ -39,49 +47,62 @@ const RANDOM_CLOSEST_DISTANCE: f64 = 1.0 / 1000.0;
 /// time window. The goal of this is to select the best connections over time
 /// from incoming join requests.
 pub(crate) struct TopologyManager {
-    slow_connection_evaluator: connection_evaluator::ConnectionEvaluator,
-    fast_connection_evaluator: connection_evaluator::ConnectionEvaluator,
-    request_density_tracker: request_density_tracker::RequestDensityTracker,
+    limits: Limits,
+    meter: Meter,
+    source_creation_times: DashMap<AttributionSource, Instant>,
+    slow_connection_evaluator: ConnectionEvaluator,
+    fast_connection_evaluator: ConnectionEvaluator,
+    request_density_tracker: RequestDensityTracker,
+    pub(crate) outbound_request_counter: OutboundRequestCounter,
     /// Must be updated when new neightbors are discovered.
     cached_density_map: CachedDensityMap,
-    pub this_peer_location: Location,
+    connection_acquisition_strategy: RwLock<ConnectionAcquisitionStrategy>,
```

Comment:
same as the `source_creation_times`field, or any inner data type of the TopologyManager, sicne you will always hold a & or a &mut dpending on the lock you shouldn't need inner mutability